src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -30,78 +34,236 @@ public MilestonePickerState(List<PickerMilestone> milestones, String userInput)
+                .forEach(milestone -> {
+                    if (milestone.isMatching()) milestone.setMatching(isMatchingQuery(milestone, queryWord));
+                });
     }
 
     /**
-     * Toggles the milestone that has milestoneName as its exact name
-     * <p>
-     * This method is case-sensitive
+     * Toggles the selection status of the milestone that has milestoneName as its exact name
      *
+     * This method is case-sensitive
      * @param milestoneName
      */
     public final void toggleExactMatchMilestone(String milestoneName) {
-        Optional<PickerMilestone> onlyMatchingMilestone = getExactMatchMilestone(currentMilestonesList, milestoneName);
-        onlyMatchingMilestone.ifPresent(milestone -> toggleMilestone(milestone));
+        Optional<PickerMilestone> exactMatchMilestone = getExactMatchMilestone(allMilestones, milestoneName);
+        exactMatchMilestone.ifPresent(this::toggleMilestone);
     }
 
     /**
-     * Toggles the milestone that is the first match to the given query
-     * <p>
-     * This method is NOT case-sensitive
+     * Toggles the selection status of the first milestone in the list
      *
      * @param milestoneList
      */
-    private final void toggleFirstMatchingMilestone(List<PickerMilestone> milestoneList) {
-        Optional<PickerMilestone> firstMatchingMilestone = getFirstMatchingMilestone(milestoneList);
+    private final void toggleFirstMilestone(List<PickerMilestone> milestoneList) {
+        Optional<PickerMilestone> firstMatchingMilestone = milestoneList.stream()
+                                                            .findFirst();
         firstMatchingMilestone.ifPresent(this::toggleMilestone);
     }
 
     /**
-     * Changes the selection statuses of all milestones in the milestone list,
-     * such that the new given milestone is toggled, and the rest are not selected
+     * Changes the selection statuses of all milestones in the allMilestones,
+     * such that the given milestone is toggled, and the rest are not selected
      *
      * @param milestone
      */
     private void toggleMilestone(PickerMilestone milestone) {
-        this.currentMilestonesList.stream()
-                .forEach(listMilestone ->
-                        listMilestone.setSelected(listMilestone.equals(milestone) && !listMilestone.isSelected()));
+        allMilestones.stream()
+                .forEach(listMilestone -> listMilestone.setSelected(listMilestone.equals(milestone)
+                        && !listMilestone.isSelected()));
+    }
+
+    public List<PickerMilestone> getAllMilestones() {
+        return allMilestones;
+    }
+
+    public List<PickerMilestone> getBestMatchingMilestones() {
+        return bestMatchingMilestones;
     }
 
-    public List<PickerMilestone> getCurrentMilestonesList() {
-        return this.currentMilestonesList;
+    /**
+     * Generates the best matching milestones which will then be added to bestMatchingMilestones
```

Comment:
Maybe `Populates bestMatchingMilestones from list of milestones which best matches user input`?